### PR TITLE
📖 ClusterClass: fix YAMLs in proposal

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
I'm aware that we can't keep those YAMLs up-to-date forever, but I almost used the wrong YAML (deployments => machineDeployments) based on the proposal.

I fixed that issue and also upgraded to v1beta1 (machineHealthCheck only exists there) to avoid confusion for readers.
(Intellij at least now doesn't report any issues in those YAMLs anymore)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/proposals/202105256-cluster-class-and-managed-topologies.md [hunk 1]
```diff
@@ -752,15 +752,15 @@ intentionally use resources without patches and variables to focus on the simple
 ##### Create a new Cluster using ClusterClass object
 1. User creates a ClusterClass object.
    ```yaml
-    apiVersion: cluster.x-k8s.io/v1alpha4
+    apiVersion: cluster.x-k8s.io/v1beta
```
Comment (fabriziopandini) [inline docs/proposals/202105256-cluster-class-and-managed-topologies.md:755]:
```suggestion
    apiVersion: cluster.x-k8s.io/v1beta1
```

**Review by killianmuldoon — COMMENTED** [review]
Can you add MachineInfrastructure to the ControlPlane? As is the create fails because you can't define a MachineHealthCheck if no MachineInfrastructure is defined.

File: docs/proposals/202105256-cluster-class-and-managed-topologies.md (hunk 1 above)
Comment (sbueringer) [reply docs/proposals/202105256-cluster-class-and-managed-topologies.md:755]: Thx!

**Comment by sbueringer** [comment]
> Can you add MachineInfrastructure to the ControlPlane? As is the create fails because you can't define a MachineHealthCheck if no MachineInfrastructure is defined.

Good point, thx. Fixed

File: docs/proposals/202105256-cluster-class-and-managed-topologies.md [hunk 2]
```diff
@@ -752,17 +752,22 @@ intentionally use resources without patches and variables to focus on the simple
...
+    apiVersion: cluster.x-k8s.io/v1beta1
     kind: ClusterClass
     metadata:
       name: mixed
       namespace: bar
     spec:
       controlPlane:
         ref:
-          apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+          apiVersion: controlplane.cluster.x-k8s.io/v1beta
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply docs/proposals/202105256-cluster-class-and-managed-topologies.md:763]: :/ Thx, I'll take a look

[the remark this answers is not in the document]
Comment (sbueringer) [reply docs/proposals/202105256-cluster-class-and-managed-topologies.md]: Should be fixed now `len(v1beta) == len(v1beta1) == 13`

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Pull request merged**